Cython/Compiler/Symtab.py

```diff
@@ -2551,6 +2551,9 @@ def declare_cfunction(self, name, type, pos,
         if self.parent_type.is_final_type or entry.is_inline_cmethod or self.directives.get('final'):
             entry.is_final_cmethod = True
             entry.final_func_cname = entry.func_cname
+            if not type.is_fused:
```

Comment:
I'm a little suspicious of excluding fused types, but I think there are probably other issues with fused methods (https://github.com/cython/cython/pull/5931) so maybe we should worry too much right now and sort it out later

Reply:
I had to exclude them here because otherwise at C code generation time, the fused type is asked for the C signature, which leads to a explicit compiler failure. 
I believe addressing fused methods in full should be for another rodeo, indeed.